### PR TITLE
Document typed arrays in optionsresolver

### PR DESCRIPTION
**Pull request by EVODelavega**
Documentation for changes introduced by [Symfony PR 17032](https://github.com/symfony/symfony/pull/17032)

### CODE REVIEW

File: components/options_resolver.rst [hunk 1]
```diff
@@ -367,6 +367,43 @@ to add additional allowed types without erasing the ones already set.
...
+        // ...
+        public function configureOptions(OptionsResolver $resolver)
+        {
+            // ...
+            $resolver->setAllowedTypes('dates', 'DateTime[]');
+            $resolver->setAllowedTypes('ports', 'int[]');
+        }
+    }
+
+Because the OptionsResolver will validate typed arrays recurively, it is possible to
```
Comment (ogizanagi) [inline components/options_resolver.rst]:
```diff
- recurively
+ recursively
```

File: components/options_resolver.rst [hunk 2]
```diff
@@ -367,6 +367,43 @@ to add additional allowed types without erasing the ones already set.
     the values to be given as an array mapping option names to allowed types:
     ``$resolver->setAllowedTypes(array('port' => array('null', 'int')));``
 
+You can specify an array of a specific type as an allowed option. The expected type is
+validated in the same way as before (``is_<type>()`` or an ``instanceof`` check).
+Only for an array, this is done recursively. If you expect an option to be an array of
```
Comment (ogizanagi) [inline components/options_resolver.rst]: Are those implementation details needed in the doc ? :/

Comment (EVODelavega) [inline components/options_resolver.rst]: @ogizanagi Well the existing documentation does mention the validation is done using the `is_<type>` functions, but we could leave it out here as an _"implied implementation detail"_ or something.

File: components/options_resolver.rst [hunk 3]
```diff
@@ -367,6 +367,43 @@ to add additional allowed types without erasing the ones already set.
...
+resolve multi-dimensional arrays, too::
+
+    // ...
+    class Mailer
+    {
+        // ...
+        public function configureOptions(OptionsResolver $resolver)
+        {
+            // ...
+            //allowed type is a 2D array of string values
```
Comment (OskarStark) [inline components/options_resolver.rst]: would add a space between ``//`` and ``allowed``

File: components/options_resolver.rst [hunk 4]
```diff
@@ -367,6 +367,43 @@ to add additional allowed types without erasing the ones already set.
...
+            $resolver->setAllowedTypes('hosts', 'string[][]');
+        }
+    }
+
+.. versionadded:: 3.1
+    Before Symfony 3.1, the allowed types had to be scalar values, qualified classes
+    or interfaces. The only way to ensure the values of an array were of the right type
```
Comment (OskarStark) [inline components/options_resolver.rst]: the last sentence sounds weird

**Comment by wouterj** [comment]
Hi @EVODelavega! Congratz on your feature!

As you submitted this doc PR 2 years ago, I've finished it for you in https://github.com/symfony/symfony-docs/pull/8946 . I've kept your commit, so that you'll still get the credits you deserve.

Thanks for starting this PR and creatign this great feature!

**Pull request closed without merging**